src/fastapibootcamp/handlers/astroplan/endpoints.py

```diff
@@ -0,0 +1,148 @@
+"""Path operations for the astroplan router."""
+
+from typing import Annotated
+
+from fastapi import APIRouter, Depends, Path, Query
+from safir.models import ErrorLocation
+
+from fastapibootcamp.dependencies.requestcontext import (
+    RequestContext,
+    context_dependency,
+)
+from fastapibootcamp.exceptions import ObserverNotFoundError
+
+from .models import (
+    ObservabilityResponseModel,
+    ObservationRequestModel,
+    ObserverModel,
+)
+
+astroplan_router = APIRouter(tags=["astroplan"])
+
+# The core of a RESTful API is the "resource". In this API, observing sites are
+# the resource. This first endpoint lets the user retrieve an existing
+# observing site resource by its ID.
+#
+# e.g. GET /astroplan/observers/rubin
+
+# We can declare path parameters that are used commonly across multiple
+# endpoints in a single place.
+
+observer_id_type = Annotated[
+    str,
+    Path(
+        ...,
+        title="The observer's site ID.",
+        examples=["rubin", "rubin-aux", "gemini-north"],
+    ),
+]
+
+
+@astroplan_router.get(
+    "/observers/{observer_id}",
+    summary="Get an observer by site ID.",
+    response_model=ObserverModel,
+)
+async def get_observer(
+    observer_id: observer_id_type,
+    context: Annotated[RequestContext, Depends(context_dependency)],
+) -> ObserverModel:
+    # Use the request context and factory patterns to get an observer service.
+    factory = context.factory
+    observer_service = factory.create_observer_service()
+
+    try:
+        # Run a method on the observer service to get the observer
+        observer = await observer_service.get_observer_by_id(observer_id)
+    except ObserverNotFoundError as e:
+        # The service, and components it calls, can raise exceptions because
+        # the user input is invalid. To provide a meaningful error response,
+        # the handler augments the exception with information about what
+        # user input caused the error.
+        #
+        # The model for this error message is declared on the router in
+        # main.py.
+        e.location = ErrorLocation.path
+        e.field_path = ["observer_id"]
+        raise
+
+    # Transform the domain object into a response object.
+    response_data = ObserverModel.from_domain(
+        observer=observer, request=context.request
+    )
+    # Make any modifications to the response headers or status code.
+    context.response.headers["Location"] = response_data.self_url
```

Comment:
I thought `Location` was only defined for 201 and 3xx responses, but this seems to be returning 200. I think I'm missing something? Maybe it's just an example of something that you could do if you were returning a redirect?